src/tests/ftest/util/ior_test_base.py

```diff
@@ -385,3 +387,33 @@ def verify_pool_size(self, original_pool_info, processes):
             self.fail(
                 "Pool Free Size did not match: actual={}, expected={}".format(
                     actual_pool_size, expected_pool_size))
+
+    def execute_cmd(self, cmd, fail_on_err=True, display_output=True):
+        """Execute cmd using general_utils.pcmd
+
+          Args:
+            cmd (str): String command to be executed
+            fail_on_err (bool): Boolean for whether to fail the test if command
+                                execution returns non zero return code.
+            display_output (bool): Boolean for whether to display output.
+        """
+        try:
+            # execute bash cmds
+            self.ret = pcmd(
+                self.hostlist_clients, cmd, verbose=display_output, timeout=300)
+            if 0 not in self.ret:
+                error_hosts = NodeSet(
+                    ",".join(
+                        [str(node_set) for code, node_set in
+                         self.ret.items() if code != 0]))
+                if fail_on_err:
+                    raise CommandFailure(
```

Comment:
Why throw an exception and handle it separately ? Why not log the error and fail here ?

Reply:
That is already done in the block right below. But this block is only for the case when we don't want to fail the test in case of a failure. But still want to raise an exception. That will give flexibility to the developer to use raised exception however he/she wishes to manipulate.